File: python/code_examples/07-errors/main.js
function mayFail(x) {
  if (x < 0) {
    throw new Error("negative");
  }
  return x * 2;
}

function main() {
  try {
    console.log(mayFail(3));
    console.log(mayFail(-1));
  } catch (e) {
    console.log("caught:", e.message);
  } finally {
    console.log("cleanup");
  }
}

main();

File: python/code_examples/07-errors/main.py
def may_fail(x: int) -> int:
    if x < 0:
        raise ValueError("negative")
    return x * 2


def main() -> None:
    try:
        print(may_fail(3))
        print(may_fail(-1))
    except ValueError as e:
        print("caught:", e)
    finally:
        print("cleanup")


if __name__ == "__main__":
    main()
